APMrover2/AP_MotorsUGV.cpp

```diff
@@ -464,36 +470,41 @@ void AP_MotorsUGV::output_regular(bool armed, float ground_speed, float steering
 }
 
 // output for omni style frames
-void AP_MotorsUGV::output_omni(bool armed, float steering, float throttle)
+void AP_MotorsUGV::output_omni(bool armed, float steering, float throttle, float lateral)
 {
-    if (!is_omni_rover()) {
+    if (!has_lateral_control()) {
         return;
     }
     if (armed) {
-        // scale throttle and steering to a 1000 to 2000 range for motor throttle calculations
-        const float scaled_throttle = (throttle - (100)) * (2000 - 1000) / (-100 - (100)) + 1000;
-        const float scaled_steering = (steering - (-4500)) * (2000 - 1000) / (4500 - (-4500)) + 1000;
+        // scale throttle, steering and lateral to -1 ~ 1
+        const float scaled_throttle = throttle / 100.0f;
+        const float scaled_steering = steering / 4500.0f;
+        const float scaled_lateral = lateral / 100.0f;
 
         // calculate desired vehicle speed and direction
-        // 1500 is a place-holder value for lateral movement input
-        const float magnitude = safe_sqrt((scaled_throttle*scaled_throttle)+(1500*1500));
-        const float theta = atan2f(scaled_throttle,1500);
+        const float magnitude = safe_sqrt(sq(scaled_throttle)+sq(scaled_lateral));
+        const float theta = atan2f(scaled_throttle,scaled_lateral);
 
-        // calculate X and Y vectors using the following the equations: vx = cos(?) * magnitude and vy = sin(?) * magnitude
+        // calculate X and Y vectors using the following the equations: vx = cos(theta) * magnitude and vy = sin(theta) * magnitude
         const float Vx = -(cosf(theta)*magnitude);
         const float Vy = -(sinf(theta)*magnitude);
 
-        // calculate output throttle for each motor and scale it back to a -100 to 100 range
-        // calculations are done using the following equations: MOTOR1 = �vx, MOTOR2 = 0.5 * v � v(3/2) * vy, MOTOR 3 = 0.5 * vx + v(3/2) * vy
+        // calculate output throttle for each motor
+        // calculations are done using the following equations: MOTOR1 = -vx, MOTOR2 = 0.5 * v - v(3/2) * vy, MOTOR 3 = 0.5 * vx + v(3/2) * vy
```

Comment:
Comment is out-of-date and superfluous.

An explanation of why this maths works would be much better.  So, for example, it isn't clear to me at a glance why `float motor_1 = ...` is generally within the range -1 to 1.